src/parser.ml

```diff
@@ -1719,13 +1719,13 @@ let rec inline defs st =
     | exception Fail ->
         junk st;
         loop (Left_bracket kind :: text acc) st
-  and loop acc st =
+  and loop ~seen_link acc st =
```

Comment:
We can avoid passing this `seen_link` argument every time by storing of the `Url` delimiter `[` or `]` is active or not. That's what I did in this branch: https://github.com/ocaml/omd/compare/master...fix-519

Reply:
Let me know which solution do you prefer :)

Reply:
I'm ok with this approach! I assume you also prefer it since it's what you used for the PR :)